File: LeetCode/Easy/1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution(object):
    def tribonacci(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0:
            return 0
        if n < 3:
            return 1
        first = 0
        second = 1
        third = 1
        for i in range(3, n+1):
            temp = first + second + third
            first = second
            second = third
            third = temp
        return third

File: LeetCode/Easy/1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.go
func tribonacci(n int) int {
    if n == 0 { return 0 }
    if n < 3 { return 1}
    first := 0
    second := 1
    third := 1
    for i := 3; i < n + 1; i++ {
        temp := first + second + third
        first = second
        second = third
        third = temp
    }
    return third
}